Frontend/RecRock-Calendar-Frontend/src/app/calendar/components/login/login.component.ts

```diff
@@ -40,4 +48,11 @@ export class LoginComponent implements OnInit {
         });
     }
   }
+
+  containsCharactersValidator(characterRe: RegExp): ValidatorFn {
```

Comment:
I would create a new file for the validation functions (like a class) so it can be reused and make it closer to SOLID principles